drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -244,16 +244,15 @@ static int pwm_nrf5_sw_init(struct device *dev)
  * channels 14-15 are used by BLE controller.
  */
 static const struct pwm_config pwm_nrf5_sw_0_config = {
-#if defined(CONFIG_SOC_SERIES_NRF51X)
+#if (DT_NORDIC_NRF_SW_PWM_SW_PWM_TIMER_INSTANCE == 1)
 	.timer = NRF_TIMER1,
-	.ppi_base = 7,
-#else
+#elif (DT_NORDIC_NRF_SW_PWM_SW_PWM_TIMER_INSTANCE == 2)
```

Comment:
What about

`.timer = NRF_TIMER ## DT_NORDIC_NRF_SW_PWM_SW_PWM_TIMER_INSTANCE`

?

For easier porting to new platforms.

Reply:
As you mentioned; #include <misc/util.h>, UTIL_CAT(NRF_TIMER, DT_NORDIC_NRF_SW_PWM_SW_PWM_TIMER_INSTANCE) will work for this purpose.
But, leaving as is for now unless someone objects

Reply:
I support @SebastianBoe's suggestion. This will be easier in maintenance.
`_CONCAT` seems to be a standard Zephyr's macro for this purpose.

Reply:
_CONCAT works. Changed